pbrd/pbr_map.c

```diff
@@ -291,6 +291,8 @@ void pbr_map_policy_interface_update(const struct interface *ifp, bool state_up)
 	 */
 	for (ALL_LIST_ELEMENTS_RO(pbrm->seqnumbers, node, pbrms))
 		for (ALL_LIST_ELEMENTS_RO(pbrm->incoming, inode, pmi))
+			if (pbrms->vrf_unchanged || pbrms->vrf_lookup
```

Comment:
Could you do a call to `pbr_map_sequence_check_valid()` instead?

Or call `pbr_map_check()` above maybe?


Secondly, should you be checking validity of the map if its an interface down event? (state_up == false)

Then we might not uninstall the rule if the interface goes down and we think the pbr map is bad now.

Reply:
That's not the point of this PR @sworleys All this is changing is the check to see if we will have an encoding issue or not sending down and preventing the send down of a rule that is not encoded correctly because it does not make sense to send it down yet.  Future interface events will put us in the right state.

Reply:
Ack i understand what you mean with the second half of my statement.

But don't you think it would be better to call pbr_map_sequence_check_valid() here rather than the long if statement? It does functionally the same thing you are trying to do here: make sure the PBR map sequence you are trying to install has a valid thing its pointing to (table installed for instance). Does using that break something that I am missing?